### ISSUE TITLE
Apply lowercase normalizer to certain keyword fields in template

### ISSUE THREAD

**Issue by andrewkroh** [elastic/ecs#260]
By applying a [lowercase normalizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/normalizer.html) to keyword fields the value will be lowercased in indexing time as well as at search-time. This helps if, for example, an application writes `network.type` as `IPv4` or if a user queries for `IPV4`.

I think it makes sense to apply this to fields where case-sensitivity doesn't matter.

**Comment by webmat** [comment]
Yes, we're considering suggesting this as an option in the definitions of  #253 and #245, to add leeway for people who prefer to avoid lowercasing their values in _source. Offering tooling to help people get this done will happen after Beta 2, however.

**Comment by praseodym** [comment]
I think a lowercase normalizer should be the default for many more ECS fields: at least usernames, hostnames, domain names, MAC addresses, hashes should be queried in a case-insensitive manner.

**Comment by webmat** [comment]
Thanks for the input. We introduced the concept to a limited number of fields for beta 2, but you're right that we should tighten the screw on more fields.

Let us know if you think of more you're like to see normalized.

**Comment by ypid-geberit** [comment]
> I think a lowercase normalizer should be the default for many more ECS fields: at least usernames, hostnames, domain names, MAC addresses, hashes should be queried in a case-insensitive manner.

I agree so I added a workaround to do this to our ECS component template build process: https://github.com/geberit/elastic-ecs
